doc/tutorials/notifications.md

```diff
@@ -0,0 +1,327 @@
+Preface
+ ===
+ 
+ **The features described in this document are not
+implemented yet or experimental. This document contains a preview of
+how developers would use the notification feature**
+ 
+ This document
+explains how notifications are implemented in Elektra and how they
+can be used by application developers.
+ 
+ Notifications - Overview &
+Concept
+ ===
+ 
+ Elektra's notification feature consists of several
+components.
+ While notification features are implemented by plugins,
+applications use APIs provided by wrappers in order to use different
+plugins.
+ 
+ A wrapper for notifications provides the API for receiving
+and handling notifications.
+ A wrapper for I/O operations allows
+asynchronous notification processing by compatible plugins.
+ The
+I/O-wrapper consists of an *interface* used by transport plugins
+and a *binding* which implements the actual I/O management functions
+for a specific I/O management library.
+ 
+ Transport plugins exchange
+notifications via different protocols like D-Bus, Redis and ZeroMQ.
+ For
+each type of transport there are two types of plugins: One for sending
+and one for receiving notifications.
+ Developers do not interact with
+those plugins directly.
+ The underlying transports are transparent
+to them.
+ An internal-notification plugin implements notification
+handling functions and feeds back configuration changes from within
+the application.
+ 
+ ![Overview](../images/notifications.svg)
+ 
+ When a
+configuration key is changed Elektra can generate change notifications
+that allow applications to process those changes.
+ Developers can choose
+whether and how they want to receive and handle those notifications
+but not whether notifications are sent.
+ How notifications are sent is
+specified in the *notification configuration* by the system operator.
+
+ Notification configuration
+ ===
+ 
+ System operators can mount the
+desired transport plugins and configure them (e.g. set channel, host,
+port and credentials) either globally or when mounting a configuration
+file.
+ 
+ They need to mount both sending and receiving plugins in order
+to use a transport.
+ 
+```sh
+kdb mount file.ini system/example ini dbussend dbusreceive
```

Comment:
dbussend currently called dbus (incompatible change?)

Reply:
Not necessarily. I was thinking about adding either `dbussend` or `dbus` (and in that case changing the plugin's name) to the `infos/provides` field of the plugin contract.